src/utils/service_builder.rs

```diff
@@ -185,7 +203,15 @@ fn get_provider(config: &ProviderConfig, key_id_manager: KeyIdManager) -> Provid
                         ))
                 .with_user_pin(config.user_pin.clone())
                 .build()
-                )
+                ))
+        }
+        #[cfg(not(all(feature = "mbed-crypto-provider", feature = "pkcs11-provider")))]
+        _ => {
+            warn!(
```

Comment:
I know we're not ready for a release yet, but this sounds like a case where we should panic instead of just printing a warning - if the config file defines a provider but that provider isn't enabled, the admin might be expecting some functionality to be available which will not be there

Reply:
> we should panic instead of just printing a warning

That is what I did first but having it return an `Option` allows us to do the following things:

* even if one of the providers could not be instantiated, Parsec can still run with the other ones
* an admin might reload the configuration with a bad one, adding a provider that is not in the binary: in that case, it will print a warning without the whole service shutting down
* I added a feature in this commit that checks if no providers could be instantiated in which case the Parsec binary will exit with an error code: I think that is still a better way to exit than panicking which can be confusing.

IMO returning always seems cleaner than panicking when it is possible.

> the admin might be expecting some functionality to be available which will not be there

You have a point there: I think it is better to solve this with the admin reading the logs and executing tests after Parsec has been installed than just panicking. I guess both options are possible and we will have to make a decision about this.

Reply:
I'm ok with this going in like this, we'll have to see if we get any feedback about it